pandas/core/arrays/timedeltas.py

```diff
@@ -137,25 +139,67 @@ def dtype(self):
     # Constructors
     _attributes = ["freq"]
 
-    @classmethod
-    def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
-        # `dtype` is passed by _shallow_copy in corner cases, should always
-        #  be timedelta64[ns] if present
-        assert dtype == _TD_DTYPE
-        assert isinstance(values, np.ndarray), type(values)
+    def __init__(self, values, dtype=_TD_DTYPE, freq=None, copy=False):
+        if isinstance(values, (ABCSeries, ABCIndexClass)):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            if freq is None:
+                freq = values.freq
+            elif freq and values.freq:
+                freq = to_offset(freq)
+                freq, freq_infer = dtl.validate_inferred_freq(
+                    freq, values.freq,
+                    freq_infer=False
+                )
+            values = values._data
+
+        if not isinstance(values, np.ndarray):
+            msg = (
+                "Unexpected type '{}'. 'values' must be a TimedeltaArray "
+                "ndarray, or Series or Index containing one of those."
+            )
+            raise ValueError(msg.format(type(values).__name__))
 
         if values.dtype == 'i8':
-            values = values.view('m8[ns]')
-
-        assert values.dtype == 'm8[ns]'
-
-        result = object.__new__(cls)
-        result._data = values
-        result._freq = freq
-        return result
+            # for compat with datetime/timedelta/period shared methods,
+            #  we can sometimes get here with int64 values.  These represent
+            #  nanosecond UTC (or tz-naive) unix timestamps
+            values = values.view(_TD_DTYPE)
+
+        if values.dtype != _TD_DTYPE:
+            msg = (
```

Comment:
In seqeunce_to_td64ns we raise with a slightly different message.  Can these be shared?

Reply:
Sure. That raises a TypeError. I seem to recall most of these `_validate` methods raising ValueErrors, though I  may be wrong. Do we have a preference?